tests/lightning_tensor/test_gates_and_expval.py

```diff
@@ -0,0 +1,341 @@
+    qml.ISWAP(wires=[wires[4], wires[5]])
+    qml.PSWAP(params[0], wires=[wires[6], wires[7]])
+    qml.adjoint(qml.SISWAP(wires=[wires[0], wires[1]]))
+    qml.SISWAP(wires=[wires[4], wires[5]])
+    qml.SQISW(wires=[wires[1], wires[0]])
+    qml.CSWAP(wires=[wires[2], wires[4], wires[5]])
+    qml.Toffoli(wires=[wires[0], wires[1], wires[2]])
+    qml.CY(wires=[wires[0], wires[2]])
+    qml.CZ(wires=[wires[1], wires[3]])
+    qml.PhaseShift(params[1], wires=wires[2])
+    qml.ControlledPhaseShift(params[2], wires=[wires[0], wires[5]])
+    qml.RX(params[3], wires=wires[0])
+    qml.RY(params[4], wires=wires[1])
+    qml.RZ(params[5], wires=wires[3])
+    qml.Rot(params[6], params[7], params[8], wires=wires[0])
+    qml.CRX(params[9], wires=[wires[1], wires[0]])
+    qml.CRY(params[10], wires=[wires[3], wires[2]])
+    qml.CRZ(params[11], wires=[wires[2], wires[1]])
+    qml.IsingXX(params[12], wires=[wires[1], wires[0]])
+    qml.IsingYY(params[13], wires=[wires[3], wires[2]])
+    qml.IsingXY(params[14], wires=[wires[2], wires[1]])
+    qml.IsingZZ(params[15], wires=[wires[2], wires[1]])
+    qml.SingleExcitation(params[16], wires=[wires[2], wires[0]])
+    qml.SingleExcitationPlus(params[17], wires=[wires[3], wires[1]])
+    qml.SingleExcitationMinus(params[18], wires=[wires[4], wires[2]])
+    qml.DoubleExcitation(params[19], wires=[wires[0], wires[1], wires[2], wires[3]])
+    qml.QubitCarry(wires=[wires[0], wires[1], wires[6], wires[7]])
+    qml.QubitSum(wires=[wires[2], wires[3], wires[7]])
+    qml.OrbitalRotation(params[20], wires=[wires[0], wires[1], wires[5], wires[6]])
+    qml.QFT(wires=[wires[0]])
+    qml.ECR(wires=[wires[1], wires[3]])
+    qml.BlockEncode([[0.1, 0.2], [0.3, 0.4]], wires=[wires[0], wires[3]])
+    qml.ctrl(qml.BlockEncode([0.1], wires=[wires[0]]), control=wires[1])
+
+
+@pytest.mark.parametrize(
+    "returns",
+    [
+        (qml.PauliX(0),),
+        (qml.PauliY(0),),
+        (qml.PauliZ(0),),
+        (qml.PauliX(1),),
+        (qml.PauliY(1),),
+        (qml.PauliZ(1),),
+        (qml.PauliX(2),),
+        (qml.PauliY(2),),
+        (qml.PauliZ(2),),
+        (qml.PauliX(3),),
+        (qml.PauliY(3),),
+        (qml.PauliZ(3),),
+        (qml.PauliX(0), qml.PauliY(1)),
+        (
+            qml.PauliZ(0),
+            qml.PauliX(1),
+            qml.PauliY(2),
+        ),
+        (
+            qml.PauliY(0),
+            qml.PauliZ(1),
+            qml.PauliY(3),
+        ),
+        (qml.PauliZ(0) @ qml.PauliY(3),),
+        (qml.Hadamard(2),),
+        (qml.Hadamard(3) @ qml.PauliZ(2),),
+        (qml.PauliX(0) @ qml.PauliY(3),),
+        (qml.PauliY(0) @ qml.PauliY(2) @ qml.PauliY(3),),
+        (qml.PauliZ(0) @ qml.PauliZ(1) @ qml.PauliZ(2),),
+        (0.5 * qml.PauliZ(0) @ qml.PauliZ(2),),
+        (qml.ops.LinearCombination([1.0, 2.0], [qml.X(0) @ qml.Z(1), qml.Y(3) @ qml.Z(2)])),
+        (qml.ops.prod(qml.X(0), qml.Y(1))),
+    ],
+)
+def test_integration_for_all_supported_gates(returns):
```

Comment:
If this test is only for `LightningTensor` then it should be skipped if `device_name != "lightning.tensor"`